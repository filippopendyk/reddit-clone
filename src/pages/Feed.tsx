import { useParams } from "react-router-dom";
import { useEffect, useState } from "react";
import { useAppDispatch } from "../hooks";

const Feed: React.FC = () => {
    let { category } = useParams();

    return <h2>This is feed component with {category ? category : 'Home'} category!</h2>
}


export default Feed;