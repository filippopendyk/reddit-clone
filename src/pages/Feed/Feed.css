main{
    display: flex;
    align-items: center;
    justify-content: center;
}

.wrapper{
    width: 100%;
    height: 100vh;
}

@media only screen and (min-width: 768px) {
    .wrapper{
        max-width: 60%;
    }
}