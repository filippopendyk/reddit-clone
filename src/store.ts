import { configureStore } from "@reduxjs/toolkit";
import filterSlice from "./features/filter/filterSlice";
import { postsSliceReducer } from "./features/posts/postsSlice";
import { fetchMostPopularCategories, topCategoriesReducer } from "./features/topCategories/topCategoriesSlice";

const store = configureStore({
    reducer: {
        topCategories: topCategoriesReducer,
        posts: postsSliceReducer,
        filter: filterSlice,
    },
});

store.dispatch(fetchMostPopularCategories());

export default store;

// Define types for usedispatch and useselector

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch;