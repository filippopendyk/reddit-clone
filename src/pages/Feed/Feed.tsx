import { useParams } from "react-router-dom";
import { useEffect, useState } from "react";
import { useAppDispatch } from "../../hooks";
import { fetchPosts } from "../../features/posts/postsSlice";
import Posts from "../../containers/Posts/Posts";
import { useAppSelector } from "../../hooks";
import FilterMenu from "../../components/FilterMenu/FilterMenu";
import './Feed.css';

const Feed: React.FC = () => {
    let { category } = useParams();
    const dispatch = useAppDispatch();

    const { data, isLoading, error } = useAppSelector((state) => state.posts);

    useEffect(() => {
        let topic = category ? category : 'all';

        dispatch(fetchPosts(topic));
    },[category]);

    return (
        <main>
            <div className="wrapper">
                <FilterMenu/>
                <Posts data={data} isLoading={isLoading} error={error}/>
            </div>
        </main>
    )
}


export default Feed;