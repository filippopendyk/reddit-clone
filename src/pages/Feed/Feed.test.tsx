import Feed from "./Feed";
import renderer from 'react-test-renderer';
import { Provider } from "react-redux";
import store from "../../store";

describe('Feed', () => {
    it('Renders without crashing', () => {
        const tree = renderer
        .create(
            <Provider store={store}>
                <Feed />
            </Provider>)
        .toJSON();
        expect(tree).toMatchSnapshot();
    })
})