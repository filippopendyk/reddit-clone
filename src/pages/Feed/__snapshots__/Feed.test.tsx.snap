// Vitest Snapshot v1

exports[`Feed > Renders without crashing 1`] = `
<main>
  <div
    className="wrapper"
  >
    <div
      className="filter-menu"
    >
      <button
        className="activeFilter"
        onClick={[Function]}
      >
        Best
      </button>
      <button
        onClick={[Function]}
      >
        Top
      </button>
      <button
        onClick={[Function]}
      >
        New
      </button>
    </div>
    <div
      className="posts-container"
    >
      <ul />
    </div>
  </div>
</main>
`;
